lib/kubernetes-deploy/container_logs.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class ContainerLogs
+    attr_reader :lines, :container_name
+
+    DEFAULT_LINE_LIMIT = 250
+
+    def initialize(parent_id:, container_name:, logger:)
+      @parent_id = parent_id
+      @container_name = container_name
+      @logger = logger
+      @lines = []
+      @last_printed_index = -1
+    end
+
+    def sync(kubectl)
```

Comment:
just out of curiosity, why is `kubectl` not an instance variable passed via the constructor?

Reply:
Because the objects instantiating this only get `kubectl` instances passed to them in certain methods, to make it a lot harder to do the wrong thing and make API calls / change internal state at unexpected points in the polling cycle (which could cause really confusing bugs). We _could_ make each of those methods do an `@logs ||=` itself instead of sharing the instance via an accessor, but that feels weirder/messier to me. And it would let them sync their logs at arbitrary times.

Reply:
makes sense 👍